RealmSwift/Tests/ObjectTests.swift

```diff
@@ -806,31 +806,56 @@ class ObjectTests: TestCase {
         XCTAssertEqual(data, object.dataCol)
     }
 
+    // MARK: - Observation tests
+
     func testObserveUnmanagedObject() {
         assertThrows(SwiftIntObject().observe { _ in }, reason: "managed")
+        assertThrows(SwiftIntObject().observe(keyPaths: ["intCol"]) { _ in }, reason: "managed")
     }
 
     func testDeleteObservedObject() {
         let realm = try! Realm()
         realm.beginWrite()
-        let object = realm.create(SwiftIntObject.self, value: [0])
+        let object0 = realm.create(SwiftIntObject.self, value: [0])
+        let object1 = realm.create(SwiftIntObject.self, value: [0])
         try! realm.commitWrite()
 
-        let exp = expectation(description: "")
-        let token = object.observe { change in
+        let exp0 = expectation(description: "Delete observed object")
+        let token0 = object0.observe { change in
             if case .deleted = change {
             } else {
                 XCTFail("expected .deleted, got \(change)")
             }
-            exp.fulfill()
+            exp0.fulfill()
+        }
+
+        let exp1 = expectation(description: "Delete observed object")
+        let token1 = object1.observe(keyPaths: ["intCol"]) { change in
+            if case .deleted = change {
```

Comment:
I know it's done this way in other tests, but this `if else` should probably be a `guard`.

Reply:
Is it okay to return from a test? 
Since `guard`s require no fall-through, this would have to return or throw. But both seem not quite right for a test. I think _almost_ every time we use guard in a test we have a fatalError in `else` -- though this usually test setup/precondition stuff.

But there is one in ModernKVOTests that returns. So I put `return` for now.

Reply:
If the guard case fails, it should `XCFail` with a message and then return.